client/snap_op.go

```diff
@@ -39,6 +39,8 @@ type SnapOptions struct {
 	Dangerous        bool   `json:"dangerous,omitempty"`
 	IgnoreValidation bool   `json:"ignore-validation,omitempty"`
 	Unaliased        bool   `json:"unaliased,omitempty"`
+
+	Homes []string `json:"homes,omitempty"`
```

Comment:
This is exactly what they are. The user might disappear before, during or after the snapshot is saved, but their data might still be important (or not).

Reply:
That sounds a bit unexpected. If the user disappears after or during the snapshot this is irrelevant.. we'll resolve the username to their home on request. It feels like making the API more annoying to use for the very exceptional case of trying to snapshot data from someone that doesn't exist in the system. I'd comfortably fail in those cases instead.

Reply:
We talked about this some more (in a standup? on IRC? not sure) and the conclusion was that I'd rejigger things so that we stored usernames in the snapshot (we also touched on whether tar stored uids or usernames, and as long as `--numeric-owner` is not given, and the user exists, it stores usernames, so the worry about uid mismatch goes away).